src/Nethermind/Nethermind.Wallet/NodeKeyManager.cs

```diff
@@ -72,16 +72,18 @@ public PrivateKey LoadNodeKey()
             // this is not secure at all but this is just the node key, nothing critical so far, will use the key store here later and allow to manage by password when launching the node
             if (_config.TestNodeKey == null)
             {
-                string oldPath = UnsecuredNodeKeyFilePath;
-                string newPath = Path.Combine(_config.KeyStoreDirectory, UnsecuredNodeKeyFilePath);
+                var executingDirectory = PathUtils.GetExecutingDirectory();
+                string oldPath = Path.Combine(executingDirectory, UnsecuredNodeKeyFilePath);
+                string newPath = Path.Combine(executingDirectory, _config.KeyStoreDirectory, UnsecuredNodeKeyFilePath);
                 
                 if (!File.Exists(newPath))
                 {
                     if (_logger.IsInfo) _logger.Info("Generating private key for the node (no node key in configuration) - stored in plain + key store for JSON RPC unlocking");
                     PrivateKey nodeKey = File.Exists(oldPath) ? new PrivateKey(File.ReadAllBytes(oldPath)) : new PrivateKeyGenerator(_cryptoRandom).Generate();
-                    if (!Directory.Exists(_config.KeyStoreDirectory))
+                    var keyStoreDirectory = Path.Combine(executingDirectory, _config.KeyStoreDirectory);
+                    if (!Directory.Exists(keyStoreDirectory))
```

Comment:
don't have to check if directory exists, just use CreateDirectory it will only create if its not there.